### PR TITLE
TestRegistrationMultiAMF 

### PR DESCRIPTION
**Pull request by JiMMyMatrix**
Add:
1. config / multiAMF repo
2. registration_test.go / TestRegistrationMultiAMF

Modify:
1. test.sh

### CODE REVIEW

**Review by brianchennn — APPROVED** [review]

File: test/registration_test.go [hunk 1]
```diff
@@ -3692,3 +3693,322 @@ func TestEAPAKAPrimeAuthentication(t *testing.T) {
...
+	_, err = conn.Write(sendMsg)
+	assert.Nil(t, err)
+
+	time.Sleep(1000 * time.Millisecond)
+
+	conn.Close()
+
+	// -----Second registration----- ue re-registration with GUTI
+
+	GUTI5GS := nasType.MobileIdentity5GS{
```
[the remark this answers is not in the document]
Comment (YouShengLiu) [reply test/registration_test.go:3871]:
```
// receive ngap Initial Context Setup Request Msg
// ...
	var guti *nasType.GUTI5G
	for _, ie := range ngapPdu.InitiatingMessage.Value.InitialContextSetupRequest.ProtocolIEs.List {
		if ie.Id.Value == ngapType.ProtocolIEIDNASPDU {
			payload := []byte(ie.Value.NASPDU.Value)
			m, err := test.NASDecode(ue, nas.GetSecurityHeaderType(payload), payload)
			assert.Nil(t, err)
			guti = m.RegistrationAccept.GUTI5G
		}
	}
// ...
	// -----Second registration----- ue re-registration with GUTI

	GUTI5GS := nasType.MobileIdentity5GS{
		Iei:    guti.Iei,
		Len:    guti.Len,
		Buffer: guti.Octet[:],
	}
```

File: test/registration_test.go [hunk 2]
```diff
@@ -3692,3 +3693,325 @@ func TestEAPAKAPrimeAuthentication(t *testing.T) {
...
+		Iei:    guti.Iei,
+		Len:    guti.Len,
+		Buffer: guti.Octet[:],
+	}
+
+	// Send registration request to new amf
+	innerRegistrationRequest := nasTestpacket.GetRegistrationRequest(nasMessage.RegistrationType5GSInitialRegistration,
```
Comment (ianchen0119) [inline test/registration_test.go:3888]: @JiMMyMatrix 
The registration request, be sent to new AMF, should be integrity protected if UE has its own security context

Comment (JiMMyMatrix) [reply test/registration_test.go:3887]: Ciphering inner registration with NEA2 and change both amf config's cipheringOrder into NEA2

**Review by ianchen0119 — APPROVED** [review]

**Comment by tim-ywliu** [comment]
@JiMMyMatrix please rebase and fix conflicts and update amf hash to re-test

**Pull request closed without merging**